### PR TITLE
PXC-3364 : Make MTR workers configurable for param jobs

### PR DESCRIPTION
**Pull request by rameshvs02**
Issue : 
          Currently, PXC-57/80 MTR pipeline jobs do not have the option to set custom MTR workers

Solution:
          Added the following parameters to set custom MTR workers
          PARALLEL_RUN : This value is used for all test suites except for Galera specific suites.
          GALERA_PARALLEL_RUN : This value is used for the Galera specific test suites
          MTR_SUITES : mysql-test-run.pl suite names
          FULL_MTR : Run MTR suites based on variable MTR_SUITES if the value is `no`. Otherwise, the full MTR will be performed.

PXC-5.7 test run result: https://pxc.cd.percona.com/job/test-pxc-5.7-param
PXC-8.0 test run result: https://pxc.cd.percona.com/job/test-pxc-8.0-param
          
PS : Removed debian jessie from the OS list as it has reached EOL

### CODE REVIEW

File: pxc/local/test-binary-pxc [hunk 1]
```diff
@@ -96,17 +65,137 @@ if [[ "${DEFAULT_TESTING}" != "no" ]]; then
...
-        ${MTR_ARGS} \
-        --force \
-        --max-test-fail=0 \
-        --suite-timeout=9999 \
-        --testcase-timeout=${TESTCASE_TIMEOUT} \
-        | tee ${WORKDIR_ABS}/mtr.output || status=$?
-    process_mtr_output PXC "${WORKDIR_ABS}/mtr.output" "${WORKDIR_ABS}/junit.xml"
+    if [ ${FULL_MTR} == "no" ]; then
+        # galera suite
+        if echo "$MTR_SUITES" | grep -Ewq "galera"; then
```
Comment (dutow) [inline pxc/local/test-binary-pxc:70]: This implementation will ignore any non default value (innodb, gr, x, main, ...) specified to the MTR_SUITES argument, which is a bad idea: as jenkins only report failures, this implies that the entire suite succeeded, while in truth it wasn't even ran.

Please make sure that anything specified by the user in these configuration values will be correctly used by the script.

Comment (rameshvs02) [reply pxc/local/test-binary-pxc]: Updated MTR script to support non-default MTR suites. Custom configurations will be used correctly through MTR_ARGS.

File: pxc/docker/run-test57 [hunk 2]
```diff
@@ -22,15 +22,19 @@ docker run --rm \
     export MTR_ARGS='${MTR_ARGS}'
     export MTR_REPEAT='${MTR_REPEAT}'
     export MTR_VAULT_TOKEN='${MTR_VAULT_TOKEN}'
+	export FULL_MTR='${FULL_MTR}'
```
Comment (dutow) [inline pxc/docker/run-test57:25]: Please fix formatting issues: previous lines used 4 spaces for indentation, these now use tabs

File: pxc/docker/run-test57 [hunk 3]
```diff
@@ -22,15 +22,19 @@ docker run --rm \
...
     mkdir /tmp/results
     cp /tmp/pxc/results/*.tar.gz /tmp/results
 
     bash -x /tmp/scripts/test-binary-pxc57 /tmp/results
 
     mkdir /tmp/results/log
-    rsync -a --prune-empty-dirs --include '*/' --include '*.err' --exclude '*' /tmp/results/PXC/mysql-test/var /tmp/results/log
-    sudo tar -C /tmp/results/log -czf "pxc57-test-mtr_logs.tar.gz" var
+    rsync -a --prune-empty-dirs --include '*/' --include '*.err' --exclude '*' /tmp/results/PXC/mysql-test/mtr_var /tmp/results/log
+    sudo tar -C /tmp/results/log -czf "pxc57-test-mtr_logs.tar.gz" mtr_var || true
```
Comment (dutow) [inline pxc/docker/run-test57:37]: Why isn't it an error if we can't create an archive? Please add a comment about it, as it's not trivial

File: pxc/jenkins/pxc57-pipeline.groovy [hunk 4]
```diff
@@ -23,7 +23,7 @@ pipeline {
             name: 'PXB24_BRANCH',
             trim: true)
         choice(
-            choices: 'centos:6\ncentos:7\ncentos:8\nubuntu:xenial\nubuntu:bionic\nubuntu:focal\ndebian:jessie\ndebian:stretch\ndebian:buster',
+            choices: 'centos:6\ncentos:7\ncentos:8\nubuntu:xenial\nubuntu:bionic\nubuntu:focal\ndebian:stretch\ndebian:buster',
```
Comment (dutow) [inline pxc/jenkins/pxc57-pipeline.groovy:26]: This change (and a few similar, like pxb versions) is not related to the commit message. Are you sure we should include then in this PR?

File: pxc/local/test-binary-pxc [hunk 5]
```diff
@@ -96,17 +65,58 @@ if [[ "${DEFAULT_TESTING}" != "no" ]]; then
...
-        --max-test-fail=0 \
-        --suite-timeout=9999 \
-        --testcase-timeout=${TESTCASE_TIMEOUT} \
-        | tee ${WORKDIR_ABS}/mtr.output || status=$?
-    process_mtr_output PXC "${WORKDIR_ABS}/mtr.output" "${WORKDIR_ABS}/junit.xml"
+    mkdir mtr_var
+    if [ ${FULL_MTR} == "no" ]; then
+        ARRAY_MTR_SUITES=($(echo $MTR_SUITES | sed 's/,/ /g'))
+        for suite in "${ARRAY_MTR_SUITES[@]}"; do
+	        if [[ "$suite" == *"galera"* ]]; then
```
Comment (dutow) [inline pxc/local/test-binary-pxc:72]: What about galera_sr, galera_3nodes, galera_3nodes_sr, ... ?

**Review by dutow — COMMENTED** [review]
I've added some minor comments.

Also, I still see one possible design issue with this: currently on 5.6 and 5.7, we already run the full mtr suite by default, and we plan to do the same for 8.0. And this diff won't help the galera suites in that configuration, only if we don't run the full mtr suite.
The default also looks to be the non full mtr, which is not what we want: by default we should run all suites, to ensure that we don't break anything.

File: pxc/docker/run-test57 (hunk 2 above)
Comment (rameshvs02) [reply pxc/docker/run-test57]: done

File: pxc/docker/run-test57 (hunk 3 above)
Comment (rameshvs02) [reply pxc/docker/run-test57]: It was added for testing purposes, removed `|| true` from archive creation.

File: pxc/jenkins/pxc57-pipeline.groovy (hunk 4 above)
Comment (rameshvs02) [reply pxc/jenkins/pxc57-pipeline.groovy:26]: Debian Jessie has reached EOL, so thought of removing it. I think we include this in the PR, will update the PR comment.

File: pxc/local/test-binary-pxc (hunk 5 above)
Comment (rameshvs02) [reply pxc/local/test-binary-pxc:72]: This condition will also work with other Galera suites.

**Review by dutow — APPROVED** [review]

**Pull request merged**